solr/core/src/java/org/apache/solr/cloud/SolrZkServer.java

```diff
@@ -277,8 +276,8 @@ public static boolean hasServers(Properties props) {
     return false;
   }
 
-  public void setDataDir(File dataDir) {
-    this.dataDir = dataDir;
+  public void setDataDir(Path dataDir) {
+    this.dataDir = dataDir.toFile();
```

Comment:
Yeah this one is setting `dataDir` from an import existing in `org.apache.zookeeper.server.quorum.QuorumPeerConfig` which is outside of Solr. If you see `toFile()` it is most likely because it either got out of hand and I put `TODO` to come back later or it wasn't possible to use `Path` strictly